### ISSUE TITLE
config.schema.json wrong defaults

### ISSUE THREAD

**Issue by woernsi** [seydx/homebridge-camera-ui#361]
Hi,

thanks for this great tool! till date, used FFMPEG plugin for my cameras to be integrated into homekit, now with HSV in BETA 1.4.0, tried out your plugin and it's great!

In total, I work with 3 Cameras, of which 2 work under RTSP and integrate easily. Remaining camera operates with MJPEG and this cannot be changed in my setup. FFMPEG allows for this integration and it worked ok on there.

When I try to enter the MJPEG camera into camera.ui on Homebridge, the camera won't surface properly but throw errors on Camera.ui & Homebridge log, speficially regaring -stimeout and -rtsp_transport.

Took a look at the config.json with the assumption that these entries would simply not be recognizes for MJPEG streams so wanted to delete them so they would not be tried to read. That worked for me (so delting -stimeout and -rtsp_transport).

In the plugin config though, rtsp_transport is a mandatory field and -stimeout automatically repopulates with 10.
Would it make sense to make these entries conditional to RTSP streams only and for MJPEG (which works with FFMPEG), do not make these entries to the config.json?

Let me know if you need more/if this helps. Again, great work!

**Environment**
- Node.js Version: v16.13.1
- Homebridge Version: v1.4.0. beta 7

- Camera UI Version: v5.0.9
- Operating System: Raspbian

**Comment by seydx** [comment]
Hey yes post your config.json pls

**Comment by seydx** [comment]
> Hi,
> 
> thanks for this great tool! till date, used FFMPEG plugin for my cameras to be integrated into homekit, now with HSV in BETA 1.4.0, tried out your plugin and it's great!
> 
> In total, I work with 3 Cameras, of which 2 work under RTSP and integrate easily. Remaining camera operates with MJPEG and this cannot be changed in my setup. FFMPEG allows for this integration and it worked ok on there.
> 
> When I try to enter the MJPEG camera into camera.ui on Homebridge, the camera won't surface properly but throw errors on Camera.ui & Homebridge log, speficially regaring -stimeout and -rtsp_transport.
> 
> Took a look at the config.json with the assumption that these entries would simply not be recognizes for MJPEG streams so wanted to delete them so they would not be tried to read. That worked for me (so delting -stimeout and -rtsp_transport).
> 
> In the plugin config though, rtsp_transport is a mandatory field and -stimeout automatically repopulates with 10. Would it make sense to make these entries conditional to RTSP streams only and for MJPEG (which works with FFMPEG), do not make these entries to the config.json?
> 
> Let me know if you need more/if this helps. Again, great work!
> 
> **Environment**
> 
> * Node.js Version: v16.13.1
> * Homebridge Version: v1.4.0. beta 7
> * Camera UI Version: v5.0.9
> * Operating System: Raspbian

these fields arent required, if you dont need these field, set `"stimeout": 0` in config and remove `"rtspTransport"` from config

**Comment by woernsi** [comment]
Hi seydx,

true and that's what I did - however when you do the configuration **through the homebridge UI** and not directly in the config.json:
- RTSP Transport shows as mandatory (has the '*') next to the field and overrides my manual deletion if I refresh with the UI config
- Socket Timeout seems to override my manual deletion in the config as well. Trying to save with an empty field will add "                        "stimeout": 10," to the config

In short, using the ui to config vs. directly editing in config.json will let the values resurface. and then cause the issues again.

**Comment by seydx** [comment]
> Hi seydx,
> 
> true and that's what I did - however when you do the configuration **through the homebridge UI** and not directly in the config.json:
> 
> * RTSP Transport shows as mandatory (has the '*') next to the field and overrides my manual deletion if I refresh with the UI config
> * Socket Timeout seems to override my manual deletion in the config as well. Trying to save with an empty field will add "                        "stimeout": 10," to the config
> 
> In short, using the ui to config vs. directly editing in config.json will let the values resurface. and then cause the issues again.

Ah you re right, thank you for the report, i will fix it :)

**Comment by woernsi** [comment]
And thanks to you for the great work :)

**Comment by seydx** [comment]
fixed in v5.0.10-beta.5

**Comment by seydx** [comment]
v5.0.10 is online
